doc/hardware/porting/board_porting.rst

```diff
@@ -8,17 +8,76 @@ directory* with various files in it. Files in the board directory inherit
 support for at least one SoC and all of its features. Therefore, Zephyr must
 support your :term:`SoC` as well.
 
+.. _board_and_identifiers:
+
+Board and board identifiers
+***************************
+
+A board may be a physical piece of hardware or an emulated board.
+Furthermore a board may contain one or multiple SoCs. Also, each SoC may contain
+one or multiple cpuclusters. Note, a cpucluster may refer to a single CPU core.
+
+It's possible to define variants for dedicated use-cases.
```

Comment:
> I wonder if variants should be moved to their own section, after revisions.

I think that could be a valid follow-up, for now I believe it's more important to have it as part of the board identifier, as it tied to soc / cpucluster in how it's specified and sanitized into file names.

> it would've been nice to have a precise definition of a variant

tricky, because a variant is very generic and is intended to cover all the (mis)use-case where variants in v1 was (wrongly) done as an independent board, simply because a proper model wasn't available.

So variant is quite generic, and covers cases not identified by soc / cpucluster.

Will consider how to phrase, but if you have a concrete proposal then I would be happy to see it.

> what distinguishes it from a revision

I think I would prefer to not introduce / reference revision at this point.
But could make sense to describe this in revision section.

Especially because a revision can be seen as a special case of a variant.